src/services/httpparsing.cpp

```diff
@@ -168,257 +75,13 @@ static void parsePostUrlEncoded(Parameters &params, std::istream &in, int conten
-
-		std::string& params_str = pars["content-type"];
-		std::map<std::string, std::string> cparams;
-
-		while ((cur = params_str.find_first_of(";", last)) != std::string::npos) {
-			std::string sub = params_str.substr(last, cur - last);
-			if (last > 0) {
-				parseKeyValuePair(sub, cparams, true);
-			}
-			last = cur + 1;
-		}
-
-		if (last < params_str.length()) {
-			std::string sub = params_str.substr(last, params_str.length() - last);
-			parseKeyValuePair(sub, cparams, true);
-		}
-
-		if (cparams.find("boundary") != cparams.end()) {
-			std::string boundary = cparams["boundary"];
-			trim(boundary, false, true); // RFC specifies only right whitespace removal.
-			if (boundary.length() >= 70) // RFC explicitly restricts this to 70 characters.
-				throw ArgumentException("Boundary length exceeded.");
-
-			std::stringstream bodyStream;
-			bodyStream << body;
-			body.clear();
-			parseMultipartBoundary(params, bodyStream, boundary);
-
-			return;
-
-		}
-
-	}
-
-	// If this is not a multipart message treat it as any regular post message (that may contain form data as well).
-
-	const std::string& content_disposition = pars["content-disposition"];
-	if(!content_disposition.empty()) {
-		std::string::size_type last = 0;
-		std::string::size_type cur;
-
-		std::map<std::string, std::string> cdparams;
-		std::string type;
-
-		while ((cur = content_disposition.find_first_of(";", last)) != std::string::npos) {
-			std::string sub = content_disposition.substr(last, cur - last);
-			if(last == 0) {
-				type = sub;
-				std::transform(type.begin(), type.end(), type.begin(), tolower);
-			}
-			else
-				parseKeyValuePair(sub, cdparams, true, "=");
-			last = cur + 1;
-		}
-
-		if (last < content_disposition.length()) {
-			std::string sub = content_disposition.substr(last, content_disposition.length() - last);
-			parseKeyValuePair(sub, cdparams, true, "=");
-		}
-
-		if(type == "form-data") {
-
-			if(cdparams["name"].empty()) {
-				throw ArgumentException("Missing form-data name parameter.");
-			}
-
-			std::string name = cdparams["name"];
-/*
-			if(!cdparams["filename"].empty()) {
-
-				// Extend HTTPService::Params with information about where to save (/tmp by default?)
-				const std::string directory = Configuration::get("multipart_default_file_directory", "/tmp");
-				std::string filename = directory + "/" +cdparams["filename"];
-
-				std::ofstream of(filename);
-				if(!of.good()) {
-					throw ArgumentException("Cannot open file " + filename + " for writing");
-				}
-
-				of << body;
-				of.close();
-
-				params[name] = filename;
-			}
-			else */{
-				// Store file in memory (string encoded into HTTPService::Params? What about binary data?)
-				params[name] = body;
-
-			}
-
-		}
-		else {
-			throw ArgumentException("Unsupported content disposition type.");
-		}
-	}
-
-}
-
 /**
- * Parses a multipart POST request
+ * Get Reader for multipart data.
+ * Parsing multipart post data is not supported right now.
```

Comment:
what do you mean by this? It is parsed by Poco, just not converted to Parameters, isn't it?

Reply:
It may be the wrong choice of words. The MultipartReader parses the data. I meant that accessing the wanted pieces of data from the MultipartReader takes a few steps and that this work isn't provided by the http parsing.